closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,486 @@
+
+# The index.html, options, and packages files
+OPENJ9_JVM_INDEX_FILE := $(OPENJ9_JVM_DOCDIR)/index.html
+OPENJ9_JVM_OPTIONS_FILE := $(DOCSTMPDIR)/openj9.jvm.options
+OPENJ9_JVM_PACKAGES_FILE := $(DOCSTMPDIR)/openj9.jvm.packages
+
+openj9-jvmdocs: $(OPENJ9_JVM_INDEX_FILE)
+
+# Set relative location to platform document root
+$(OPENJ9_JVM_INDEX_FILE): GET2DOCSDIR := $(PLATFORM_DOCSDIR)/..
+
+# Run javadoc if the index file is out of date or missing
+$(OPENJ9_JVM_INDEX_FILE): $(OPENJ9_JVM_OPTIONS_FILE) $(OPENJ9_JVM_PACKAGES_FILE)
+	$(prep-javadoc)
+	$(call JavadocSummary,$(OPENJ9_JVM_OPTIONS_FILE),$(OPENJ9_JVM_PACKAGES_FILE))
+	$(JAVADOC_CMD) -d $(@D) \
+	    @$(OPENJ9_JVM_OPTIONS_FILE) @$(OPENJ9_JVM_PACKAGES_FILE)
+
+$(OPENJ9_JVM_OPTIONS_FILE): FIRST_COPYRIGHT_YEAR := $(OPENJ9_JVM_FIRST_COPYRIGHT_YEAR)
+$(OPENJ9_JVM_OPTIONS_FILE):
+	$(prep-target)
+	@($(call OptionOnly,$(COMMON_JAVADOCFLAGS)) ; \
+	  $(call OptionPair,-sourcepath,$(OPENJ9_DOCS_SRCDIR)) ; \
+	  $(call OptionPair,-encoding,ascii) ; \
+	  $(call OptionPair,-doctitle,$(OPENJ9_JVM_DOCTITLE)) ; \
+	  $(call OptionPair,-windowtitle,$(OPENJ9_JVM_WINDOWTITLE) $(DRAFT_WINTITLE)) ; \
+	  $(call OptionPair,-header,$(OPENJ9_JVM_HEADER)$(DRAFT_HEADER)) ; \
+	  $(call OptionPair,-bottom,$(OPENJ9_JAVADOC_BOTTOM)$(DRAFT_BOTTOM)) ; \
+	  $(call OptionTrip,-linkoffline,$(JAVASE_BASE_URL),$(COREAPI_DOCSDIR)/) ; \
+	) > $@
+
+# Create a file with the package names in it
+$(OPENJ9_JVM_PACKAGES_FILE): $(DIRECTORY_CACHE) $(call PackageDependencies,$(OPENJ9_JVM_PKGS))
+	$(prep-target)
+	$(call PackageFilter,$(OPENJ9_JVM_PKGS))
+
+#############################################################################
+#
+# openj9 mgmtdocs
+#
+# The openjdk com.sun.management classes are specific to the
+# hotspot virtual machine and do not work with openj9.
+# openj9 provides the com.ibm.lang.management,
+# com.ibm.virtualization.management and
+# openj9.lang.management packages for managing the openj9
+# virtual machine.
+# Some of the openj9 com.ibm.lang.management classes implement
+# com.sun.management interfaces, so those com.sun.management
+# classes need to be included in the javadoc.
+#
+# The required openjdk com.sun.management interface source files
+# are copied to a temporary location, which is then used as the
+# source file location for the javadoc, rather than the actual
+# openjdk source tree.
+#
+# The MGMT_XXX macros are defined in make/Javadoc.gmk
+
+#############################################################################
+# Location of the openjdk com.sun.management source files
+SUN_MGMT_SRCDIR := $(JDK_SHARE_CLASSES)/com/sun/management
+
+#############################################################################
+# Somewhere to copy the openjdk com.sun.management interfaces referenced by
+# the com.ibm.lang.management to.
+TEMP_SUN_MGMT_PKGDIR := $(TEMPDIR)/mgmt
+TEMP_SUN_MGMT_SRCDIR := $(TEMP_SUN_MGMT_PKGDIR)/com/sun/management
```

Comment:
How does copying files to this directory help? I can't find a way that `javadoc` is informed of the directory.

Reply:
Copying the files to the temporary directory provides the means to exclude the jdk.management files which are not supported by an openj9 jdk.

The directory is added to the openj9 classes directory in the statement
```
OPENJ9_MGMT_SRCPATH := $(OPENJ9_DOCS_SRCDIR)$(PATH_SEP)$(TEMP_SUN_MGMT_PKGDIR)
```
and then passed to javadoc as the -sourcepath argument in the options file:
```
$(OPENJ9_MGMT_OPTIONS_FILE):
	.....
	  $(call OptionPair,-sourcepath,$(OPENJ9_MGMT_SRCPATH)) ; \
	.....
```